java/clients/src/test/java/sleeper/clients/teardown/ShutdownSystemProcessesIT.java

```diff
@@ -172,6 +187,71 @@ void shouldStopECSTaskWhenOneIsFound() throws Exception {
     @Nested
     @DisplayName("Terminate running EMR clusters")
     class TerminateEMRClusters {
+
+        @Test
+        void shouldStopEMRServerlessWhenApplicationIsStartedWithNoJobs() throws Exception {
+            //Given
+            InstanceProperties properties = createTestInstancePropertiesWithEmrStack();
+            stubFor(listActiveClustersRequest()
+                .willReturn(aResponseWithNumRunningClusters(0)));
+            stubFor(listActiveApplicationsRequest()
+                .willReturn(aResponseWithNumRunningApplications(1)));
+            stubFor(listJobsForApplicationsRequest("test-application-id-1")
+                .willReturn(aResponseWithNumRunningJobsOnApplication(0)));
+            stubFor(stopJobForApplicationsRequest("test-application-id-1")
+                .willReturn(aResponseWithNumRunningJobsOnApplication(0)));
+
+            // When
+            shutdown(properties);
+
+            // Then
+            verify(2, getRequestedFor(urlEqualTo("/applications")));
+            verify(2, listActiveApplicationRequested());
+
+        }
+
+        @Test
+        void emrServerlessHasZeroApplicationsStarted() throws Exception {
+            //Given
+            InstanceProperties properties = createTestInstancePropertiesWithEmrStack();
+            stubFor(listActiveClustersRequest()
+                .willReturn(aResponseWithNumRunningClusters(0)));
+            stubFor(listActiveApplicationsRequest()
+                .willReturn(aResponseWithNumRunningApplications(0)));
+
+            // When
+            shutdown(properties);
+
+            // Then
+            verify(1, getRequestedFor(urlEqualTo("/applications")));
+            verify(1, listActiveApplicationRequested());
+
+        }
+
+        @Test
+        void shouldStopEMRServerlessWhenApplicationIsStartedWithRunningJobs() throws Exception {
+            //Given
+            InstanceProperties properties = createTestInstancePropertiesWithEmrStack();
+            stubFor(listActiveClustersRequest()
+                .willReturn(aResponseWithNumRunningClusters(0)));
+            stubFor(listActiveApplicationsRequest()
+                .willReturn(aResponseWithNumRunningApplications(1)));
+            stubFor(listJobsForApplicationsRequest("test-application-id-1")
+                .willReturn(aResponseWithNumRunningJobsOnApplication(1)));
+            stubFor(stopJobForApplicationsRequest("test-application-id-1")
+                .willReturn(aResponseWithNumRunningJobsOnApplication(0)));
+            stubFor(deleteJobsForApplicationsRequest("test-application-id-1", "test-job-run-id-1")
+                .willReturn(ResponseDefinitionBuilder.okForEmptyJson()));
+
+            // When
+            shutdown(properties);
+
+            // Then
+            verify(2, getRequestedFor(urlEqualTo("/applications")));
+            verify(2, listActiveApplicationRequested());
+
+        }
+
```

Comment:
Please can we test the case where an application exists but is not currently running?

Reply:
Added new tests